app/javascript/mastodon/components/scrollable_list.js

```diff
@@ -209,13 +209,6 @@ export default class ScrollableList extends PureComponent {
     this.setState({ fullscreen: isFullscreen() });
   }
 
-  attachIntersectionObserver () {
-    this.intersectionObserverWrapper.connect({
-      root: this.node,
```

Comment:
Now wait... You're fixing it for single-column mode but breaking it for advanced layout, I think.

Reply:
I tested it in both modes; is there some specific functionality I missed?

Reply:
Ok, I've also discovered a discrepancy: in single-column mode, the column is of arbitrary size (extending indefinitely downwards into the browser, causing the behavior this PR is supposed to fix); whereas, in column mode, each column stops at the bottom (well, slightly above) of the viewport, and is scrollable (hideIfNotIntersecting is indeed called).  If you'd prefer, I can try to make a CSS change to make the column-layout consistent across advanced/basic interface modes (thereby hopefully achieving the same behavior for both); but I'm still currently of the opinion that the viewport is the correct root element for intersection calculation, regardless of interface.

Reply:
Well, in single-column mode, the scrollbar is on the document body itself, while in advanced layout, the scrollbar is on ScrollableList, which is why this code, which originates from when only advanced layout existed, uses `this.node` as `root`. Seems like it should be conditional as either `document.body` or `this.node` depending on layout (`this.props.bindToDocument`)

Reply:
ok, I'll try introducing the conditional and seeing what happens; will update with results

Reply:
this seems to work